### PR TITLE
Added tests for elemental analysis data file

### LINKED ISSUE #26593: Elemental analysis data file needs to be tested
There should be some tests checking the correct formatting of the peak data file.
These should test that:
 - every peak is assigned to an element
 - properties like `A`, mass, name and primary peaks exist for all elements

### PR DESCRIPTION
**Pull request by BassoLorenzo**
Added tests to check that data file do not contain obviously wrong data (eg. negative mass) and are formatted correctly.
This includes:
 - having all peak types assigned to an element (see  #26430)
 - ensuring each element has an entry for `A`, `Z`, `Primary`, `Secondary`
 - ensuring at least one element has an entry for `Gammas`, `Electrons`

**To test:**

- Code review
- Check that the test pass with the default data file
- Modify the values of some peaks to negative ones, it should fail
- Add an entry for a non existing element, it should fail
- Remove `Pb->Electrons`, it should fail
- Remove one of `A`, `Z`, `Primary`, `Secondary` from any element, it should fail

Fixes #26593.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/elemental_analysis/peak_data_file_test.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+from Muon.GUI.ElementalAnalysis.elemental_analysis import ElementalAnalysisGui
+from Muon.GUI.ElementalAnalysis.elemental_analysis import gen_name
+from MultiPlotting.multi_plotting_widget import MultiPlotWindow
+from MultiPlotting.multi_plotting_widget import MultiPlotWidget
+from MultiPlotting.label import Label
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_model import PeriodicTableModel
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table import _elements
+from Muon.GUI.ElementalAnalysis.PeriodicTable.PeakSelector.peak_selector_view import valid_data
+
```
Comment (Harrietbrown) [inline scripts/test/Muon/elemental_analysis/peak_data_file_test.py]: There are a lot of unused import here, only import what you are using.

File: scripts/test/Muon/elemental_analysis/peak_data_file_test.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            from_file = self.data.get(element[0], None)
+            if from_file is not None:
+                self.assertEqual(element[1], from_file['Z'])
+
+    def test_atomic_mass_matches_the_one_in_periodic_table(self):
+        for element in _elements:
+            from_file = self.data.get(element[0], None)
+            if from_file is not None and from_file['A'] is not None:
+                self.assertAlmostEqual(element[5], from_file['A'])
+
```
Comment (Harrietbrown) [inline scripts/test/Muon/elemental_analysis/peak_data_file_test.py]: It is good to have this in your tests so they can be ran independantly.

```suggestion


if __name__ == "__main__":
    unittest.main()

```

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
It all looks very good and the tests work fine. Just two small changes.

**Review by Harrietbrown — DISMISSED** [review]
This works well, good job.

**Comment by martyngigg** [comment]
@LolloB There are a few merge conflicts after merging #26432. Could you rebase, please?

**Review by Harrietbrown — APPROVED** [review]
Conflicts fixed, everything good.

**Pull request merged**